test/mpi/pt2pt/inactivereq.c

```diff
@@ -79,6 +79,52 @@ int test_recv_init(int src_rank, const char *test_name)
     return errs;
 }
 
+int test_proc_null()
+{
+    MPI_Request r;
+    MPI_Status s;
+    int errs = 0;
+    int flag;
+    int buf[10];
+    int tag = 27;
+    const char *test_name = "proc_null";
+
+    MPI_Recv_init(buf, 10, MPI_INT, MPI_PROC_NULL, tag, MPI_COMM_WORLD, &r);
+    MPI_Start(&r);
+    MPI_Wait(&r, &s);
+
+    MPI_Send_init(buf, 10, MPI_INT, MPI_PROC_NULL, tag, MPI_COMM_WORLD, &r);
+    MPI_Start(&r);
+    MPI_Wait(&r, &s);
+
+    flag = 0;
+    s.MPI_TAG = 10;
+    s.MPI_SOURCE = 10;
+    MPI_Test(&r, &flag, &s);
```

Comment:
You already ran `MPI_Wait` on the request, which sets the request to `MPI_REQUEST_NULL`. What's the point of `MPI_Test` and `MPI_Wait` on it again?

Reply:
Persistent requests become inactive after completion with `MPI_TEST` or `MPI_WAIT`. They are not set to `MPI_REQUEST_NULL`.